README.md

```diff
@@ -3,10 +3,39 @@ Repository of re-usable test utilities.
 
 ### Modules
 The following modules are defined:
-* module-1
+* spock-shaded
+* spock-extensions
+* definalizer
  
-#### module-1
-Defines something
+#### spock-shaded
+Defines a single jar file that contains all dependencies required to develop and execute Spock specifications.
+
+#### spock-extensions
+Defines a set of extensions that can be used when defining Spock specifications.
+##### [@ClearInterruptions](spock-extensions/src/main/groovy/org/codice/spock/extension/builtin/ClearInterruptionsExtension.groovy)
+An annotation that can be used to clear interruption states from the current thread after testing
+##### [@Supplemental](spock-extensions/src/main/groovy/org/codice/spock/extension/Supplemental.groovy) 
+An annotation that extends certain classes (e.g. java.lang.Class, java.lang.reflect.Method, java.lang.System, spock.mock.MockingApi, ...) to provide additional useful functionality.
+
+#### definalizer
+The Definalizer JUnit test runner is designed as a generic proxy test runner for another JUnit test runner by indirectly instantiating that runner in order to add support for de-finalizing (i.e. removing the final constraint) 3rd party Java classes that needs to be mocked or stubbed during testing. 
+It does so by creating a classloader designed with an aggressive strategy where it will load all classes first before delegating to its parent. This classloader will therefore reload all classes while definalizing those that are requested except for all classes in the following packages:
+* java
+* javax
+* sun
+* org.xml
+* org.junit
+
+These packages are not being reloaded as they are required for this test runner to delegate to the real test runner. Even the actual test class will be reloaded in this internal classloader.
+
+The indirect extension is done by means of delegation as the real test runner is instantiated from within the classloader that is created internally. 
+This is to ensure that everything the test class, the test runner, and everything they indirectly reference are loaded from within the classloader.
+
+This runner is especially useful with Spock where it is not possible to mock final methods as can be done with Mockito.
+
+The @Definalize annotation should be added to the test class to specify which classes and/or packages to definalize.
+
+The @DefinalizeWith annotation can be added to specify which actual test runner should be used to run the test case. By default, it wil leither use the standard JUnit or Sputnik test runner.
```

Comment:
✏️ Probably should wrap `@DefinalizeWith` in backticks.

❗️ s/b "By default, it will either..."